doc/tor.1.txt

```diff
@@ -647,6 +647,37 @@ GENERAL OPTIONS
     in accordance to RFC 1929. Both username and password must be between 1 and
     255 characters.
 
+[[TCPProxy]] **TCPProxy** __protocol__ __host__:__port__::
+	Tor will use the given protocol to make all its OR (SSL) connections through
+	a TCP proxy on host:port, rather than connecting directly to servers. You may
+	want to set **FascistFirewall** to restrict the set of ports you might try to
+	connect to, if your proxy only allows connecting to certain ports. There is no
+	equivalent option for directory connections, because all Tor client versions
+	that support this option download directory documents via OR connections. +
++
+	The only protocol supported right now 'haproxy'. This option is only for
+	clients. (Default: none) +
++
+	The haproxy protocol works in the following way: when the feature is enabled,
+	the Tor process will write a header line on every outbound connection. The header
+	is in the following format: +
++
+	"PROXY" [ "TCP4" | "TCP6" ] SourceIPAddress DestinationIPAddress SourcePort DestinationPort "\r\n" +
++
+	There is a single space after each item, except for the last item, which is
+	followed by a CRLF. +
++
+	After parsing a correctly-formatted PROXY line, the haproxy connects to
+	DestinationIPAddress:DestinationPort, and forwards all subsequent data to the
+	destination. Any data sent by the destination is forwarded by haproxy to
+	the Tor client. +
++
+	In the case that Tor cannot find the source IP address and/or source port,
+	both will be set to zero.
```

Comment:
For privacy reasons, we should always set these fields to zero.
Otherwise, clients may leak internal IP addresses.